sys/shell/shell.c

```diff
@@ -225,7 +225,7 @@ static int readline(shell_t *shell, char *buf, size_t size)
         if (c == '\r' || c == '\n') {
             if (line_buf_ptr == buf) {
                 /* The line is empty. */
-                continue;
+                return 1;
             }
 
             *line_buf_ptr = '\0';
```

Comment:
(nevermind my previous comment, that was a bug in pyterm)
moving the 

            shell->put_char('\r');
            shell->put_char('\n');

to above the if block will make the prompt appear on its own line when using minicom or similar terminal programs.